src/main/java/dev/sigstore/encryption/signers/RsaVerifier.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.encryption.signers;
+
+import java.security.*;
+
+/** RSA verifier, instantiated by {@link Verifiers#newVerifier(PublicKey)}. */
+public class RsaVerifier implements Verifier {
+
+  private final PublicKey publicKey;
+
+  RsaVerifier(PublicKey publicKey) {
+    this.publicKey = publicKey;
+  }
+
+  @Override
+  public PublicKey getPublicKey() {
+    return publicKey;
+  }
+
+  @Override
+  public boolean verify(byte[] artifact, byte[] signature)
```

Comment:
these two methods look identical apart from the algo. maybe delegate to internal shared method?

Reply:
Man I had them all as parameterized versions of the same class. But then this weirdo rsa hash prefix got me good